VocaDbWeb.Core/Middleware/VocaDbPrincipalMiddleware.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using NLog;
+using VocaDb.Model;
+using VocaDb.Model.Service;
+using VocaDb.Model.Service.Security;
+using VocaDb.Web.Code;
+
+namespace VocaDb.Web.Middleware
+{
+	public class VocaDbPrincipalMiddleware
+	{
+		private readonly RequestDelegate _next;
+
+		public VocaDbPrincipalMiddleware(RequestDelegate next)
+		{
+			_next = next;
+		}
+
+		private static bool IsAjaxRequest(HttpRequest request)
+		{
+			ParamIs.NotNull(() => request);
+
+			return request.Headers["X-Requested-With"] == "XMLHttpRequest";
+		}

```

Comment:
Needs to be tested.

Reply:
In any case this class should be temporary and eventually be replaced with something that fits ASP.NET Core authentication better.